### PR TITLE
001-Create-Skeleton

### PR DESCRIPTION
**Pull request by MorganMcDonald**
Uploading a code skeleton for the frontend

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Request, Response } from "express";
```
Comment (glenn-hamilton) [inline app.ts]: This is unused

File: app.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const express = require('express');
+const path = require('path');
+const nunjucks = require('nunjucks');
+const session =require('express-session')
+//const { title } = require('process');
```
Comment (glenn-hamilton) [inline app.ts]: This is unused (and commented out)

File: app.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+//configure Express
+app.set('view engine', 'html');
+
+app.use('/public',express.static(path.join(__dirname, 'public')));
+
+app.use(express.json())
+
+app.use(express.urlencoded({ extended: true }));
+
+app.use(session({secret: 'NOT HARDCODED SECRET', cookie: {maxAge: 60000}}));
```
Comment (glenn-hamilton) [inline app.ts]: Don't need sessions currently

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        "target": "ES5",
+        "sourceMap": true,
+        "noImplicitAny": true
+    },
+    "include": ["./**/*"],
+    "exclude": [
+        "./node_modules/**/*",
+        "./dist/**/*"
+    ]
+}
```
Comment (glenn-hamilton) [inline tsconfig.json:14]: Needs a new line at the end

File: app.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    coCache: true,
+    express: app
+};
+
+nunjucks.configure(appViews, nunjucksConfig);
+
+//configure Express
+app.set('view engine', 'html');
+
+app.use('/public',express.static(path.join(__dirname, 'public')));
```
Comment (glenn-hamilton) [inline app.ts:22]: This currently isn't testable, can you add a view and hello world route

File: app.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { Request, Response } from "express";
+const express = require('express');
+const path = require('path');
+const nunjucks = require('nunjucks');
+const session =require('express-session')
```
Comment (glenn-hamilton) [inline app.ts]: This import isn't used

File: package.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    "nunjucks": "^3.2.4"
+  },
+  "devDependencies": {
+    "@types/express": "^4.17.17",
+    "@types/node": "^20.5.7",
+    "typescript": "^5.2.2"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/omcmenamin/FunkBot-Frontend-WebAPI.git"
```
Comment (glenn-hamilton) [inline package.json]: This is wrong

File: views/helloWorld.html [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<html>
+    <p>Hello World</p>
+</html>
```
Comment (glenn-hamilton) [inline views/helloWorld.html]: New line needed

File: controller/helloWorldController.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { Application, Request, Response } from "express";
+
+const session = require ('express');
```
Comment (glenn-hamilton) [inline controller/helloWorldController.ts]: No need for a session here

File: controller/helloWorldController.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+module.exports = function(app: Application){
+
+    app.get('/helloworld', async (req:Request, res: Response)=>{
+    res.render('helloWorld')
```
Comment (glenn-hamilton) [inline controller/helloWorldController.ts]: This should probably be indented one more level

**Review by glenn-hamilton — APPROVED** [review]

**Review by chloe22simpson — APPROVED** [review]

**Pull request merged**